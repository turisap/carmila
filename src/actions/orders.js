/**
 * Created by HP on 10-Dec-17.
 */

/**
 * Adds an item to the items array
 * @param item
 */
export const addItem = (item) => ({
    type : 'ADD_ITEM',
    item,
});

/**
 * Increment counter of an item in order list
 * @param title
 */
export const incrementItem = (title) => ({
    type : 'INCREMENT_ITEM',
    title
});

/**
 * Removes an item from the items array
 * @param title
 */
export const removeItem = (title) => ({
    type : 'REMOVE_ITEM',
    title
});

/**
 * Removes all same items from order (4 same pizzas for example)
 * @param title
 */
export const removeAllSameItems = (title) => ({
    type : 'REMOVE_ALL_SAME_ITEMS',
    title
});

/**
 * Fills state with user's credentials
 * @param data
 */
export const fillCredentials = (data) => ({
    type : 'FILL_CREDENTIALS',
    data
});

/**
 * Sets pick up time for an order
 * @param time
 */
export const setPickupTime = (time) => ({
    type : 'SET_PICKUP_TIME',
    time
});

