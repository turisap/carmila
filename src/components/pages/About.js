/**
 * Created by HP on 09-Dec-17.
 */
import React from 'react';

const About = () => (
    <div>
        <h1>About page</h1>
    </div>
);

export default About;