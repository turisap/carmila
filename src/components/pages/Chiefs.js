/**
 * Created by HP on 09-Dec-17.
 */
import React from 'react';

const Chiefs = () => (
    <div className="container">
        <div className="row">
            <div className="col-md-6">

            </div>
        </div>
    </div>
);

export default Chiefs;