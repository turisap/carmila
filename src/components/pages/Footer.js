/**
 * Created by HP on 09-Dec-17.
 */
import React from 'react';
import {NavLink} from 'react-router-dom';

const Footer = () => (
    <div className="footer__fix">

        <footer>
            <ul className="footer__links">
                <li><NavLink className="footer__link text-center" to="/about">About Us</NavLink></li>
                <li><NavLink className="footer__link text-center" to="#">Sign Up For News</NavLink></li>
            </ul>
            <ul className="footer__info">
                <li>
                    <p id="footer__info">Our office is located 24, Rd No 2 Krong Battambang.
                        We are open 8a.m. to 5p.m Monday to Friday.</p>
                </li>
            </ul>
        </footer>
    </div>
);

export default Footer;