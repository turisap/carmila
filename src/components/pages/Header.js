/**
 * Created by HP on 09-Dec-17.
 */
import React from 'react';
import {NavLink} from 'react-router-dom';
import {connect} from 'react-redux';

export const Header = (props) => (
    <div>
        <nav className="navbar navbar-expand-lg navbar-light bg-light navbar-right">
            <NavLink className="navbar-brand" to="/">Caldira</NavLink>
            <div className="collapse navbar-collapse" id="navbarColor03">
                <ul className="navbar-nav">
                    <li className="nav-item">
                        <NavLink activeClassName="active" className="nav-link" to="/">Home</NavLink>
                    </li>
                    <li className="nav-item">
                        <NavLink activeClassName="active" className="nav-link" to="/dishes">Menu</NavLink>
                    </li>
                    {props.haveItemsInOrder &&
                    <li className="nav-item">
                        <NavLink activeClassName="active" className="nav-link" to="/order">Order</NavLink>
                    </li>
                    }
                </ul>
            </div>
        </nav>
    </div>
);

const mapStateToProps = (state) => ({
    haveItemsInOrder : (state.orders.items.length > 0)
});

export default connect(mapStateToProps)(Header);