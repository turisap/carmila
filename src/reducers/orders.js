/**
 * Created by HP on 10-Dec-17.
 */

const orderDefaultReducerState = {
    items : [],
    time : undefined,
    name : '',
    phone : ''
};

export default (state = orderDefaultReducerState, action) => {
    switch(action.type) {
        case 'ADD_ITEM':
            return {
                ...state,
                items : incrementSameItem(state.items, action.item)
            };
            break;
        case 'REMOVE_ITEM':
            const newState = {...state};
            newState.items.forEach((item,index,array) => {
                if (item.title === action.title) {
                    item.amount--
                }
                if (item.amount === 0) {
                    array.splice(index, 1);
                }
            });
            return newState;
            break;
        case 'INCREMENT_ITEM':
            const newStater = {...state};
            newStater.items.forEach((item,index,array) => {
                if (item.title === action.title) {
                    item.amount ? item.amount++ : item.amount = 1;
                }
            });
            return newStater;
            break;
        case 'REMOVE_ALL_SAME_ITEMS':
            return {
                ...state,
                items : state.items.filter(item => item.title !== action.title)
            };
            break;
        case 'FILL_CREDENTIALS':
            return {
                ...state,
                name : action.data.name,
                phone : action.data.phone
            };
            break;
        case 'SET_PICKUP_TIME':
            return {
                ...state,
                time : action.time,
            };
            break;
        default:
            return state;
    }
}

export const incrementSameItem = (initialItems, additionalItem) => {
    const amount = {amount : 1};
    if(initialItems.length !== 0) {
        let exist = false;
        initialItems.forEach(i => {
            if (i.title === additionalItem.title) {
                i.amount ? i.amount++ : Object.assign(i, amount);
                exist = true;
            }
        });
        return exist ? initialItems : [...initialItems, Object.assign(additionalItem, amount)];
    }
    return [Object.assign(additionalItem, amount)];
};