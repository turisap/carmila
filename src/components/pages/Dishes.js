/**
 * Created by HP on 09-Dec-17.
 */
import React from 'react';
import { connect } from 'react-redux';
import DishItem from './../DishItem';
import visibleDishes from '../../selectors/visibleDishes';
import getRowsOfDishes from '../../adapters/chunk';
import DishesFilterForm from '../DishesFilterForm';
import { addItem, removeItem, fillCredentials } from '../../actions/orders';


export class Dishes extends React.Component {
    handleAddButton = (item) => {
        this.props.addItemToOrder(item)
    };

    getAddedTitles = () => {
        return [...this.props.items].map(item => item.title);
    };

    render(){
        return (
            <div>
                <section className="dishesFilterForm">
                    <div className="container-fluid">
                        <div className="row">
                            <div className="col-md-12">
                                <DishesFilterForm/>
                            </div>
                        </div>
                    </div>
                </section>
                <section className="dishList">
                    <div className="container-fluid">
                        {getRowsOfDishes(3, this.props.dishes, false, this.handleAddButton, this.getAddedTitles())}
                    </div>
                </section>

            </div>
        )
    }
}

const mapStateToProps = (state) => ({
    dishes : visibleDishes(state.filter.filterBy, state.filter.text),
    items  : state.orders.items
});

const mapDispatchToProps = (dispatch, props) => ({
    addItemToOrder : (item) => dispatch(addItem(item)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Dishes);