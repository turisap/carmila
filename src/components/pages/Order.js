/**
 * Created by HP on 09-Dec-17.
 */
import React from 'react';
import {connect} from 'react-redux';
import { TimePicker } from 'antd';
import moment from 'moment';
import Link, {LinkedComponent} from 'valuelink';
import {Input} from 'valuelink/tags';
import {NavLink} from 'react-router-dom'
import getRowsOfDishes from '../../adapters/chunk';
import {setPickupTime, fillCredentials, removeItem} from '../../actions/orders';

export class Order extends LinkedComponent {
    state = {
        name : '',
        phone : '',
        timeError : undefined,
        submitted : false
    };

    onDateChange = (e) => {
        if (!e.isAfter(moment().add(1, 'hour'))) {
            this.setState({timeError : 'We need at least one hour to prepare you order'});
            this.props.setPickupTime(undefined);
            return;
        }
        this.setState({timeError : undefined});
        this.props.setPickupTime(e.format('Do MMM, HH:mm'));
    };

    handleOrderPlacing = (e) => {
        e.preventDefault();
        this.setState({submitted:true});
        if(!this.state.timeError) {
            this.props.fillCredentials(this.state)
        }
    };

    render () {
        const linked = this.linkAll();
        const nameLink = Link.state(this, 'name')
            .check(x => x, 'Name is required');
        const phoneLink = Link.state(this, 'phone')
            .check(x => x, 'Phone is required')
            .check(x => x.match(/^[0-9]*$/), 'Phone number should be valid');
        const emptyList = !this.props.orders.items.length > 0;

        return (
            <div>
                <div className={emptyList ? "orderList__holder--empty" : "container-fluid orderList__holder"}>
                    <section className="orderList__heading" style={{visibility: emptyList ? 'hidden' : ''}}>
                        <div className="container-fluid">
                            <h2 className="page-header">We are happy to serve food for you</h2>
                            <hr/>
                        </div>
                    </section>
                    <section className="orderList">
                        {this.props.orders.items && getRowsOfDishes(3, this.props.orders.items, true)}
                    </section>
                    { emptyList &&
                    <div className="orderList__noItems">
                        <h2>You haven't chosen anything yet</h2>
                        <h2 className="error"><NavLink to="/dishes">Take A Look To The Menu</NavLink></h2>
                    </div>
                    }
                    <section className="orderForm">
                        {(this.props.orders.items && this.props.orders.items.length > 0) &&
                        <form onSubmit={this.handleOrderPlacing} className="orderForm__form">
                            <FormInput
                                valueLink={nameLink}
                                submitted={this.state.submitted}
                                className="form-control orderForm__input"
                                placeholder="Your name"
                            />
                            <FormInput
                                valueLink={phoneLink}
                                submitted={this.state.submitted}
                                className="form-control orderForm__input"
                                placeholder="Your phone"
                            />
                            <TimePicker
                                defaultValue={moment().add(1, 'hour')}
                                format={'HH:mm'}
                                onChange={this.onDateChange}
                                size="large"
                            />
                            <button href="#" id="placeOrder" type="submit" className="btn btn-default">
                                Place Order
                            </button>
                        </form>
                        }
                        {(this.state.timeError && this.state.submitted) && <p className="error">{this.state.timeError}</p>}
                    </section>
                </div>
            </div>

        )
    }
}

const FormInput = ({...props}) => (
    <div>
        <Input {...props}/>
        <div className="error-placeholder">
            {(props.valueLink.error && props.submitted) && <p className="error">{props.valueLink.error}</p>}
        </div>
    </div>
);

const mapStateToProps = (state) => ({
    orders : state.orders,
});

const mapDispatchToProps = (dispatch, props) => ({
    setPickupTime : (time) => dispatch(setPickupTime(time)),
    fillCredentials : (data) => dispatch(fillCredentials(data)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Order);




