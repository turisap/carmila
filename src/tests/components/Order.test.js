/**
 * Created by HP on 15-Dec-17.
 */
import React from 'react';
import { shallow } from 'enzyme';
import {Order} from '../../components/pages/Order';
import orders from '../fixtures/fullorders';
import moment from 'moment';

let wrapper, setPickUpTime, fillCredentials;

beforeEach(() => {
    setPickUpTime = jest.fn();
    fillCredentials = jest.fn();
    wrapper = shallow(
    <Order
        setPickupTime={setPickUpTime}
        fillCredentials={fillCredentials}
        orders={orders}
    />);
});

test('should render order page properly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should set timeError', () => {
    let e = moment();
    wrapper.find('TimePicker').simulate('change', e);
    expect(wrapper.state('timeError').length).toBeGreaterThan(0);
    expect(wrapper).toMatchSnapshot();
});

test('should not set timeError and set pickuptime', () => {
    let e = moment().add(90, 'minutes');
    wrapper.find('TimePicker').simulate('change', e);
    expect(wrapper.state('timeError')).toBe(undefined);
    expect(setPickUpTime).toHaveBeenCalledWith(e.format('Do MMM, HH:mm'));
});

test('should fill credentials', () => {
    let e = moment().add(2, 'hours');
    e = {
        ...e,
        preventDefault : () => {}
    };
    wrapper.find('form').simulate('submit', e);
    expect(fillCredentials).toHaveBeenCalledWith(wrapper.state());
});
