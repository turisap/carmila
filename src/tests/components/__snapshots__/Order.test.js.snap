// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render order page properly 1`] = `
<div>
  <div
    key="0"
  >
    <Connect(DishItem)
      description="test description"
      orderPage={true}
      price={4}
      title="Cremeschnitte"
    />
  </div>
  <div
    key="1"
  >
    <Connect(DishItem)
      description="test description"
      orderPage={true}
      price={5.5}
      title="Linguine"
    />
  </div>
  <div
    key="2"
  >
    <Connect(DishItem)
      description="test description"
      orderPage={true}
      price={7}
      title="Margherita"
    />
  </div>
  <form
    onSubmit={[Function]}
  >
    <FormInput
      submitted={false}
      valueLink={
        e {
          "component": Order {
            "context": Object {},
            "handleOrderPlacing": [Function],
            "links": Object {
              "name": [Circular],
              "phone": e {
                "component": [Circular],
                "error": "Phone is required",
                "key": "phone",
                "value": "",
              },
              "submitted": e {
                "component": [Circular],
                "key": "submitted",
                "value": false,
              },
              "timeError": e {
                "component": [Circular],
                "key": "timeError",
                "value": undefined,
              },
            },
            "onDateChange": [Function],
            "props": Object {
              "fillCredentials": [Function],
              "orders": Object {
                "items": Array [
                  Object {
                    "amount": undefined,
                    "category": "PIZZA",
                    "description": "test description",
                    "price": 4,
                    "title": "Cremeschnitte",
                  },
                  Object {
                    "amount": undefined,
                    "category": "PASTA",
                    "description": "test description",
                    "price": 5.5,
                    "title": "Linguine",
                  },
                  Object {
                    "amount": undefined,
                    "category": "DESSERT",
                    "description": "test description",
                    "price": 7,
                    "title": "Margherita",
                  },
                ],
                "name": "",
                "phone": "",
                "time": undefined,
              },
              "setPickupTime": [Function],
            },
            "refs": Object {},
            "state": Object {
              "name": "",
              "phone": "",
              "submitted": false,
              "timeError": undefined,
            },
            "updater": Updater {
              "_callbacks": Array [],
              "_renderer": ReactShallowRenderer {
                "_context": Object {},
                "_element": <Order
                  fillCredentials={[Function]}
                  orders={
                    Object {
                      "items": Array [
                        Object {
                          "amount": undefined,
                          "category": "PIZZA",
                          "description": "test description",
                          "price": 4,
                          "title": "Cremeschnitte",
                        },
                        Object {
                          "amount": undefined,
                          "category": "PASTA",
                          "description": "test description",
                          "price": 5.5,
                          "title": "Linguine",
                        },
                        Object {
                          "amount": undefined,
                          "category": "DESSERT",
                          "description": "test description",
                          "price": 7,
                          "title": "Margherita",
                        },
                      ],
                      "name": "",
                      "phone": "",
                      "time": undefined,
                    }
                  }
                  setPickupTime={[Function]}
                />,
                "_forcedUpdate": false,
                "_instance": [Circular],
                "_newState": null,
                "_rendered": <div>
                  <div>
                    <Connect(DishItem)
                      amount={undefined}
                      description="test description"
                      orderPage={true}
                      price={4}
                      title="Cremeschnitte"
                    />
                  </div>
                  <div>
                    <Connect(DishItem)
                      amount={undefined}
                      description="test description"
                      orderPage={true}
                      price={5.5}
                      title="Linguine"
                    />
                  </div>
                  <div>
                    <Connect(DishItem)
                      amount={undefined}
                      description="test description"
                      orderPage={true}
                      price={7}
                      title="Margherita"
                    />
                  </div>
                  <form
                    onSubmit={[Function]}
                  >
                    <FormInput
                      submitted={false}
                      valueLink={[Circular]}
                    />
                    <FormInput
                      submitted={false}
                      valueLink={
                        e {
                          "component": [Circular],
                          "error": "Phone is required",
                          "key": "phone",
                          "value": "",
                        }
                      }
                    />
                    <TimePicker
                      align={
                        Object {
                          "offset": Array [
                            0,
                            -2,
                          ],
                        }
                      }
                      defaultValue={"1970-01-01T01:00:00.000Z"}
                      disabled={false}
                      disabledHours={undefined}
                      disabledMinutes={undefined}
                      disabledSeconds={undefined}
                      focusOnOpen={true}
                      format="HH:mm"
                      hideDisabledOptions={false}
                      onChange={[Function]}
                      placement="bottomLeft"
                      prefixCls="ant-time-picker"
                      transitionName="slide-up"
                    />
                    <button
                      href="#"
                      id="placeOrder"
                      type="submit"
                    >
                      Place Order
                    </button>
                  </form>
                </div>,
                "_rendering": false,
                "_updater": [Circular],
              },
            },
          },
          "error": "Name is required",
          "key": "name",
          "value": "",
        }
      }
    />
    <FormInput
      submitted={false}
      valueLink={
        e {
          "component": Order {
            "context": Object {},
            "handleOrderPlacing": [Function],
            "links": Object {
              "name": e {
                "component": [Circular],
                "error": "Name is required",
                "key": "name",
                "value": "",
              },
              "phone": [Circular],
              "submitted": e {
                "component": [Circular],
                "key": "submitted",
                "value": false,
              },
              "timeError": e {
                "component": [Circular],
                "key": "timeError",
                "value": undefined,
              },
            },
            "onDateChange": [Function],
            "props": Object {
              "fillCredentials": [Function],
              "orders": Object {
                "items": Array [
                  Object {
                    "amount": undefined,
                    "category": "PIZZA",
                    "description": "test description",
                    "price": 4,
                    "title": "Cremeschnitte",
                  },
                  Object {
                    "amount": undefined,
                    "category": "PASTA",
                    "description": "test description",
                    "price": 5.5,
                    "title": "Linguine",
                  },
                  Object {
                    "amount": undefined,
                    "category": "DESSERT",
                    "description": "test description",
                    "price": 7,
                    "title": "Margherita",
                  },
                ],
                "name": "",
                "phone": "",
                "time": undefined,
              },
              "setPickupTime": [Function],
            },
            "refs": Object {},
            "state": Object {
              "name": "",
              "phone": "",
              "submitted": false,
              "timeError": undefined,
            },
            "updater": Updater {
              "_callbacks": Array [],
              "_renderer": ReactShallowRenderer {
                "_context": Object {},
                "_element": <Order
                  fillCredentials={[Function]}
                  orders={
                    Object {
                      "items": Array [
                        Object {
                          "amount": undefined,
                          "category": "PIZZA",
                          "description": "test description",
                          "price": 4,
                          "title": "Cremeschnitte",
                        },
                        Object {
                          "amount": undefined,
                          "category": "PASTA",
                          "description": "test description",
                          "price": 5.5,
                          "title": "Linguine",
                        },
                        Object {
                          "amount": undefined,
                          "category": "DESSERT",
                          "description": "test description",
                          "price": 7,
                          "title": "Margherita",
                        },
                      ],
                      "name": "",
                      "phone": "",
                      "time": undefined,
                    }
                  }
                  setPickupTime={[Function]}
                />,
                "_forcedUpdate": false,
                "_instance": [Circular],
                "_newState": null,
                "_rendered": <div>
                  <div>
                    <Connect(DishItem)
                      amount={undefined}
                      description="test description"
                      orderPage={true}
                      price={4}
                      title="Cremeschnitte"
                    />
                  </div>
                  <div>
                    <Connect(DishItem)
                      amount={undefined}
                      description="test description"
                      orderPage={true}
                      price={5.5}
                      title="Linguine"
                    />
                  </div>
                  <div>
                    <Connect(DishItem)
                      amount={undefined}
                      description="test description"
                      orderPage={true}
                      price={7}
                      title="Margherita"
                    />
                  </div>
                  <form
                    onSubmit={[Function]}
                  >
                    <FormInput
                      submitted={false}
                      valueLink={
                        e {
                          "component": [Circular],
                          "error": "Name is required",
                          "key": "name",
                          "value": "",
                        }
                      }
                    />
                    <FormInput
                      submitted={false}
                      valueLink={[Circular]}
                    />
                    <TimePicker
                      align={
                        Object {
                          "offset": Array [
                            0,
                            -2,
                          ],
                        }
                      }
                      defaultValue={"1970-01-01T01:00:00.000Z"}
                      disabled={false}
                      disabledHours={undefined}
                      disabledMinutes={undefined}
                      disabledSeconds={undefined}
                      focusOnOpen={true}
                      format="HH:mm"
                      hideDisabledOptions={false}
                      onChange={[Function]}
                      placement="bottomLeft"
                      prefixCls="ant-time-picker"
                      transitionName="slide-up"
                    />
                    <button
                      href="#"
                      id="placeOrder"
                      type="submit"
                    >
                      Place Order
                    </button>
                  </form>
                </div>,
                "_rendering": false,
                "_updater": [Circular],
              },
            },
          },
          "error": "Phone is required",
          "key": "phone",
          "value": "",
        }
      }
    />
    <TimePicker
      align={
        Object {
          "offset": Array [
            0,
            -2,
          ],
        }
      }
      defaultValue={"1970-01-01T01:00:00.000Z"}
      disabled={false}
      focusOnOpen={true}
      format="HH:mm"
      hideDisabledOptions={false}
      onChange={[Function]}
      placement="bottomLeft"
      prefixCls="ant-time-picker"
      transitionName="slide-up"
    />
    <button
      href="#"
      id="placeOrder"
      type="submit"
    >
      Place Order
    </button>
  </form>
</div>
`;

exports[`should set timeError 1`] = `
<div>
  <div
    key="0"
  >
    <Connect(DishItem)
      description="test description"
      orderPage={true}
      price={4}
      title="Cremeschnitte"
    />
  </div>
  <div
    key="1"
  >
    <Connect(DishItem)
      description="test description"
      orderPage={true}
      price={5.5}
      title="Linguine"
    />
  </div>
  <div
    key="2"
  >
    <Connect(DishItem)
      description="test description"
      orderPage={true}
      price={7}
      title="Margherita"
    />
  </div>
  <form
    onSubmit={[Function]}
  >
    <FormInput
      submitted={false}
      valueLink={
        e {
          "component": Order {
            "context": Object {},
            "handleOrderPlacing": [Function],
            "links": Object {
              "name": [Circular],
              "phone": e {
                "component": [Circular],
                "error": "Phone is required",
                "key": "phone",
                "value": "",
              },
              "submitted": e {
                "component": [Circular],
                "key": "submitted",
                "value": false,
              },
              "timeError": e {
                "component": [Circular],
                "key": "timeError",
                "value": "We need at least one hour to prepare you order",
              },
            },
            "onDateChange": [Function],
            "props": Object {
              "fillCredentials": [Function],
              "orders": Object {
                "items": Array [
                  Object {
                    "amount": undefined,
                    "category": "PIZZA",
                    "description": "test description",
                    "price": 4,
                    "title": "Cremeschnitte",
                  },
                  Object {
                    "amount": undefined,
                    "category": "PASTA",
                    "description": "test description",
                    "price": 5.5,
                    "title": "Linguine",
                  },
                  Object {
                    "amount": undefined,
                    "category": "DESSERT",
                    "description": "test description",
                    "price": 7,
                    "title": "Margherita",
                  },
                ],
                "name": "",
                "phone": "",
                "time": undefined,
              },
              "setPickupTime": [Function],
            },
            "refs": Object {},
            "state": Object {
              "name": "",
              "phone": "",
              "submitted": false,
              "timeError": "We need at least one hour to prepare you order",
            },
            "updater": Updater {
              "_callbacks": Array [],
              "_renderer": ReactShallowRenderer {
                "_context": Object {},
                "_element": <Order
                  fillCredentials={[Function]}
                  orders={
                    Object {
                      "items": Array [
                        Object {
                          "amount": undefined,
                          "category": "PIZZA",
                          "description": "test description",
                          "price": 4,
                          "title": "Cremeschnitte",
                        },
                        Object {
                          "amount": undefined,
                          "category": "PASTA",
                          "description": "test description",
                          "price": 5.5,
                          "title": "Linguine",
                        },
                        Object {
                          "amount": undefined,
                          "category": "DESSERT",
                          "description": "test description",
                          "price": 7,
                          "title": "Margherita",
                        },
                      ],
                      "name": "",
                      "phone": "",
                      "time": undefined,
                    }
                  }
                  setPickupTime={[Function]}
                />,
                "_forcedUpdate": false,
                "_instance": [Circular],
                "_newState": Object {
                  "name": "",
                  "phone": "",
                  "submitted": false,
                  "timeError": "We need at least one hour to prepare you order",
                },
                "_rendered": <div>
                  <div>
                    <Connect(DishItem)
                      amount={undefined}
                      description="test description"
                      orderPage={true}
                      price={4}
                      title="Cremeschnitte"
                    />
                  </div>
                  <div>
                    <Connect(DishItem)
                      amount={undefined}
                      description="test description"
                      orderPage={true}
                      price={5.5}
                      title="Linguine"
                    />
                  </div>
                  <div>
                    <Connect(DishItem)
                      amount={undefined}
                      description="test description"
                      orderPage={true}
                      price={7}
                      title="Margherita"
                    />
                  </div>
                  <form
                    onSubmit={[Function]}
                  >
                    <FormInput
                      submitted={false}
                      valueLink={[Circular]}
                    />
                    <FormInput
                      submitted={false}
                      valueLink={
                        e {
                          "component": [Circular],
                          "error": "Phone is required",
                          "key": "phone",
                          "value": "",
                        }
                      }
                    />
                    <TimePicker
                      align={
                        Object {
                          "offset": Array [
                            0,
                            -2,
                          ],
                        }
                      }
                      defaultValue={"1970-01-01T01:00:00.000Z"}
                      disabled={false}
                      disabledHours={undefined}
                      disabledMinutes={undefined}
                      disabledSeconds={undefined}
                      focusOnOpen={true}
                      format="HH:mm"
                      hideDisabledOptions={false}
                      onChange={[Function]}
                      placement="bottomLeft"
                      prefixCls="ant-time-picker"
                      transitionName="slide-up"
                    />
                    <button
                      href="#"
                      id="placeOrder"
                      type="submit"
                    >
                      Place Order
                    </button>
                  </form>
                </div>,
                "_rendering": false,
                "_updater": [Circular],
              },
            },
          },
          "error": "Name is required",
          "key": "name",
          "value": "",
        }
      }
    />
    <FormInput
      submitted={false}
      valueLink={
        e {
          "component": Order {
            "context": Object {},
            "handleOrderPlacing": [Function],
            "links": Object {
              "name": e {
                "component": [Circular],
                "error": "Name is required",
                "key": "name",
                "value": "",
              },
              "phone": [Circular],
              "submitted": e {
                "component": [Circular],
                "key": "submitted",
                "value": false,
              },
              "timeError": e {
                "component": [Circular],
                "key": "timeError",
                "value": "We need at least one hour to prepare you order",
              },
            },
            "onDateChange": [Function],
            "props": Object {
              "fillCredentials": [Function],
              "orders": Object {
                "items": Array [
                  Object {
                    "amount": undefined,
                    "category": "PIZZA",
                    "description": "test description",
                    "price": 4,
                    "title": "Cremeschnitte",
                  },
                  Object {
                    "amount": undefined,
                    "category": "PASTA",
                    "description": "test description",
                    "price": 5.5,
                    "title": "Linguine",
                  },
                  Object {
                    "amount": undefined,
                    "category": "DESSERT",
                    "description": "test description",
                    "price": 7,
                    "title": "Margherita",
                  },
                ],
                "name": "",
                "phone": "",
                "time": undefined,
              },
              "setPickupTime": [Function],
            },
            "refs": Object {},
            "state": Object {
              "name": "",
              "phone": "",
              "submitted": false,
              "timeError": "We need at least one hour to prepare you order",
            },
            "updater": Updater {
              "_callbacks": Array [],
              "_renderer": ReactShallowRenderer {
                "_context": Object {},
                "_element": <Order
                  fillCredentials={[Function]}
                  orders={
                    Object {
                      "items": Array [
                        Object {
                          "amount": undefined,
                          "category": "PIZZA",
                          "description": "test description",
                          "price": 4,
                          "title": "Cremeschnitte",
                        },
                        Object {
                          "amount": undefined,
                          "category": "PASTA",
                          "description": "test description",
                          "price": 5.5,
                          "title": "Linguine",
                        },
                        Object {
                          "amount": undefined,
                          "category": "DESSERT",
                          "description": "test description",
                          "price": 7,
                          "title": "Margherita",
                        },
                      ],
                      "name": "",
                      "phone": "",
                      "time": undefined,
                    }
                  }
                  setPickupTime={[Function]}
                />,
                "_forcedUpdate": false,
                "_instance": [Circular],
                "_newState": Object {
                  "name": "",
                  "phone": "",
                  "submitted": false,
                  "timeError": "We need at least one hour to prepare you order",
                },
                "_rendered": <div>
                  <div>
                    <Connect(DishItem)
                      amount={undefined}
                      description="test description"
                      orderPage={true}
                      price={4}
                      title="Cremeschnitte"
                    />
                  </div>
                  <div>
                    <Connect(DishItem)
                      amount={undefined}
                      description="test description"
                      orderPage={true}
                      price={5.5}
                      title="Linguine"
                    />
                  </div>
                  <div>
                    <Connect(DishItem)
                      amount={undefined}
                      description="test description"
                      orderPage={true}
                      price={7}
                      title="Margherita"
                    />
                  </div>
                  <form
                    onSubmit={[Function]}
                  >
                    <FormInput
                      submitted={false}
                      valueLink={
                        e {
                          "component": [Circular],
                          "error": "Name is required",
                          "key": "name",
                          "value": "",
                        }
                      }
                    />
                    <FormInput
                      submitted={false}
                      valueLink={[Circular]}
                    />
                    <TimePicker
                      align={
                        Object {
                          "offset": Array [
                            0,
                            -2,
                          ],
                        }
                      }
                      defaultValue={"1970-01-01T01:00:00.000Z"}
                      disabled={false}
                      disabledHours={undefined}
                      disabledMinutes={undefined}
                      disabledSeconds={undefined}
                      focusOnOpen={true}
                      format="HH:mm"
                      hideDisabledOptions={false}
                      onChange={[Function]}
                      placement="bottomLeft"
                      prefixCls="ant-time-picker"
                      transitionName="slide-up"
                    />
                    <button
                      href="#"
                      id="placeOrder"
                      type="submit"
                    >
                      Place Order
                    </button>
                  </form>
                </div>,
                "_rendering": false,
                "_updater": [Circular],
              },
            },
          },
          "error": "Phone is required",
          "key": "phone",
          "value": "",
        }
      }
    />
    <TimePicker
      align={
        Object {
          "offset": Array [
            0,
            -2,
          ],
        }
      }
      defaultValue={"1970-01-01T01:00:00.000Z"}
      disabled={false}
      focusOnOpen={true}
      format="HH:mm"
      hideDisabledOptions={false}
      onChange={[Function]}
      placement="bottomLeft"
      prefixCls="ant-time-picker"
      transitionName="slide-up"
    />
    <button
      href="#"
      id="placeOrder"
      type="submit"
    >
      Place Order
    </button>
  </form>
</div>
`;
